### PR TITLE
[Page][Olana] Hackathons List

### PR DESCRIPTION
**Pull request by jrolana**
Note:
Created only dummy data for currently log in user will update after finishing user register/sign in page.

### CODE REVIEW

File: hackathons_list.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
```
Comment (ezerssss) [inline hackathons_list.php]: change title

File: hackathons_list.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Inter:wght@100..900&display=swap" rel="stylesheet">
+
+    <link href="style.css" rel="stylesheet">
+
+</head>
+
+<body>
+    <?php
+    include 'db_connector.php';
+    $currentUserID = 2;
```
Comment (ezerssss) [inline hackathons_list.php]: from session to right? pero like question lang ba para like on the same page di siya correcting or somethni GHAHAHAHA

File: hackathons_list.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        <h2>Joined Hackathons</h2>
+        <?php
+        $joinedHackathons_sql = "SELECT hackathonID FROM participates WHERE userID=$currentUserID";
+        $joined = $conn->query($joinedHackathons_sql);
+
+        if ($joined) {
+            while ($joinedHackathon = $joined->fetch_assoc()) {
+                $joinedHackathonID = $joinedHackathon["hackathonID"];
+                $hackathon_sql = "SELECT * FROM hackathons WHERE hackathonID=$joinedHackathonID";
+                $hackathon_details = $conn->query($hackathon_sql);
```
Comment (ezerssss) [inline hackathons_list.php]: can probably make this into a simpler one SQL query

```
SELECT * FROM hackathons WHERE hackathonsID IN (SELECT hackathonID FROM participates WHERE userID=currentUserID AND dateEnd > $currentDate)
```

maybe a little bit readable is:
```
$joinedHackathons_sql = SELECT hackathonID FROM participates WHERE userID=currentUserID AND dateEnd > $currentDate;

$hackathons = SELECT * FROM hackathons WHERE hackathonsID IN ($joinedHackathons_sql)
```

File: hackathons_list.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+                if ($hackathon_details) {
+                    while ($hackathon = $hackathon_details->fetch_assoc()) {
+                        if ($hackathon["dateEnd"] >= $currentDate) {
+                            echo "<li>" .
+                                "<h3>" . $hackathon["theme"] . "</h3>" .
+                                "<p>" . $hackathon["description"] . "</p>" .
+                                "<p>" . $hackathon["dateStart"] . " to " . $hackathon["dateEnd"] . "</p>" .
+                                "<button>Submit a project</button>" .
```
Comment (ezerssss) [inline hackathons_list.php]: change to a form lang para ready na siya for integration soon

File: hackathons_list.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+    <ul>
+        <h2>Available Hackathons</h2>
+        <?php
+        $availHackathons_sql = "SELECT hackathonID FROM participates WHERE userID!=$currentUserID";
+        $available = $conn->query($availHackathons_sql);
+
+        if ($available) {
+            while ($availHackathon = $available->fetch_assoc()) {
+                $availHackathonID = $availHackathon["hackathonID"];
```
Comment (ezerssss) [inline hackathons_list.php]: i think same sa previous ko na comment nga better if

```
$joinedHackathons_sql = SELECT hackathonID FROM participates WHERE userID=currentUserID;

$hackathons = SELECT * FROM hackathons WHERE hackathonsID NOT IN ($joinedHackathons_sql)
```

File: hackathons_list.php [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                $hackathon_details = $conn->query($hackathon_sql);
+
+                if ($hackathon_details) {
+                    while ($hackathon = $hackathon_details->fetch_assoc()) {
+                        if ($hackathon["dateEnd"] >= $currentDate) {
+                            echo "<li>" .
+                                "<h3>" . $hackathon["theme"] . "</h3>" .
+                                "<p>" . $hackathon["description"] . "</p>" .
+                                "<p>" . $hackathon["dateStart"] . " to " . $hackathon["dateEnd"] . "</p>" .
+                                "<button>Join</button>" .
```
Comment (ezerssss) [inline hackathons_list.php]: make into form

File: scratch.txt
Comment (ezerssss) [inline scratch.txt:1]: delete non final files in PR

File: hackathons_list.php [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        if ($available) {
+            while ($availHackathon = $available->fetch_assoc()) {
+                $availHackathonID = $availHackathon["hackathonID"];
+
+                $hackathon_sql = "SELECT * FROM hackathons WHERE hackathonID=$availHackathonID";
+                $hackathon_details = $conn->query($hackathon_sql);
+
+                if ($hackathon_details) {
+                    while ($hackathon = $hackathon_details->fetch_assoc()) {
+                        if ($hackathon["dateEnd"] < $currentDate) {
```
Comment (ezerssss) [inline hackathons_list.php]: ohhh here mo linagay ang if statement, better siguro if sa actual SQL ang checking para smaller rows ang makuha niya

```
$pastJoinedHackathons_sql = SELECT hackathonID FROM participates WHERE userID=currentUserID AND dateEnd <= $currentDate;

$hackathons = SELECT * FROM hackathons WHERE hackathonsID IN ($joinedHackathons_sql)
```

**Review by ezerssss — CHANGES_REQUESTED** [review]

File: hackathons_list.php (hunk 2 above)
Comment (jrolana) [reply hackathons_list.php]: yes for session

**Comment by ezerssss** [comment]
maybe also change ang name ng php to `home.php` kay siya man ang home page

**Review by ezerssss — APPROVED** [review]

**Pull request merged**